components/Snackbar/src/private/MDCSnackbarOverlayView.m

```diff
@@ -345,6 +353,29 @@ - (CGRect)snackbarRectInScreenCoordinates {
                       toCoordinateSpace:window.screen.coordinateSpace];
 }
 
+- (CGFloat)maximumHeight {
+  // Maximum height must be extended to include the bottom content safe area.
+  CGFloat maximumHeight = kMaximumHeight;
+#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
+  if (self.anchoredToScreenEdge) {
+    if (@available(iOS 11.0, *)) {
+      maximumHeight += self.window.safeAreaInsets.bottom;
+    }
+  }
+#endif
+  return maximumHeight;
+}
+
+- (BOOL)anchoredToScreenEdge {
+  return self.bottomOffset == 0;
+}
+
+#pragma mark - Safe Area Insets
+
+- (void)safeAreaInsetsDidChange {
+  [self triggerSnackbarLayoutChange];
```

Comment:
Since self.maximumHeightConstraint is using self.window.safeAreaInsets.bottom, don't we need to adjust it when the safe area changes?

Reply:
Fixed.